src_airflow/elt_pipeline.py

```diff
@@ -0,0 +1,245 @@
+YEARS = ["2022"]
+###############################################
+
+
+###############################################
+# Utils
+###############################################
+def create_bucket(bucket_name):
+     # Load minio config
+    cfg = load_cfg(CFG_FILE)
+    datalake_cfg = cfg["datalake"]
+
+    # Create a client with the MinIO server
+    minio_client = Minio(
+        endpoint=datalake_cfg["endpoint"],
+        access_key=datalake_cfg["access_key"],
+        secret_key=datalake_cfg["secret_key"],
+        secure=False,
+    )
+
+    # Create bucket if not exist
+    found = minio_client.bucket_exists(bucket_name=bucket_name)
+    if not found:
+        minio_client.make_bucket(bucket_name=bucket_name)
+    else:
+        print(f"Bucket {bucket_name} already exists, skip creating!")
+###############################################
+
+
+###############################################
+# Extract and Load
+###############################################
+def extract_load_to_datalake():
+    """
+        Extract data file and Load to Datalake (MinIO) at bucket 'datalake'
+    """
+    # Load minio config
+    cfg = load_cfg(CFG_FILE)
+    datalake_cfg = cfg["datalake"]
+    nyc_data_cfg = cfg["nyc_data"]
+
+    print(datalake_cfg)
+
+    # Create a client with the MinIO server
+    minio_client = Minio(
+        endpoint=datalake_cfg["endpoint"],
+        access_key=datalake_cfg["access_key"],
+        secret_key=datalake_cfg["secret_key"],
+        secure=False,
+    )
+
+    # Create bucket if not exist
+    found = minio_client.bucket_exists(bucket_name=datalake_cfg["bucket_name_1"])
+    if not found:
+        minio_client.make_bucket(bucket_name=datalake_cfg["bucket_name_1"])
+    else:
+        print(f"Bucket {datalake_cfg['bucket_name_1']} already exists, skip creating!")
+
+    for year in YEARS:
+        # Upload files
+        all_fps = glob(os.path.join(nyc_data_cfg["folder_path"], year, "*.parquet"))
+
+        for fp in all_fps:
+            print(f"Uploading {fp}")
+            minio_client.fput_object(
+                bucket_name=datalake_cfg["bucket_name_1"],
+                object_name=os.path.join(datalake_cfg["folder_name"], os.path.basename(fp)),
+                file_path=fp,
+            )
+###############################################
+
+
+###############################################
+# Transform
+###############################################
+def drop_column(df, file):
+    """
+        Drop columns 'store_and_fwd_flag'
+    """
+    if "store_and_fwd_flag" in df.columns:
+        df = df.drop(columns=["store_and_fwd_flag"])
+        print("Dropped column store_and_fwd_flag from file: " + file)
+    else:
+        print("Column store_and_fwd_flag not found in file: " + file)
+
+    return df
+
+
+def merge_taxi_zone(df, file):
+    """
+        Merge dataset with taxi zone lookup
+    """
+    df_lookup = pd.read_csv(TAXI_LOOKUP_PATH)
+
+    if "pickup_latitude" not in df.columns:
```

Comment:
This is a repeated code! I have seen this code in `src/`. Don't repeat your code, since i's very difficult to maintain 2 similar codes at the same time.

Reply:
Because the configuration between local and within Airflow has some differences, I want to separate it to make it easier to mount the src_airflow/ directory into Airflow when running airflow-docker-compose.yaml

Reply:
No, could you find a way to have one code only? Maintaining two copies can cause a lot of trouble.

Reply:
Fixed!